src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -140,11 +140,14 @@ class CollectorService(
     }
 
     private fun MessageFilter.toRootMessageFilter(): RootMessageFilter {
+        check(this.messageType.isNotEmpty()) { "Rule cannot be executed because the message filter does not contain 'message type'" }
         return RootMessageFilter.newBuilder()
-                .setMessageType(this.messageType)
-                .setComparisonSettings(this.comparisonSettings.toRootComparisonSettings())
-                .setMessageFilter(this)
-                .build()
+            .setMessageType(this.messageType).also {
+                logger.info("Rule used a message type '${this.messageType}' from the message filter")
```

Comment:
Why do we need this logging here? We will be able to see that in the original request